spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsFunctionProcessor.java

```diff
@@ -251,9 +258,19 @@ public void orchestrateFunctionInvoking(ResolvableType resolvableType, String fu
 					KafkaStreamsConsumerProperties extendedConsumerProperties =
 							this.kafkaStreamsExtendedBindingProperties.getExtendedConsumerProperties(input);
 					//get state store spec
-					Serde<?> keySerde = this.keyValueSerdeResolver.getInboundKeySerde(extendedConsumerProperties);
-					Serde<?> valueSerde = this.keyValueSerdeResolver.getInboundValueSerde(
-							bindingProperties.getConsumer(), extendedConsumerProperties);
+
+					Serde<?> keySerde;
+					Serde<?> valueSerde;
+
+					if (bindingServiceProperties.getConsumerProperties(input).isUseNativeDecoding()) {
+						keySerde = this.keyValueSerdeResolver.getInboundKeySerde(extendedConsumerProperties, stringResolvableTypeMap.get(input));
+						valueSerde = this.keyValueSerdeResolver.getInboundValueSerde(
+								bindingProperties.getConsumer(), extendedConsumerProperties, stringResolvableTypeMap.get(input));
+					}
+					else {
+						keySerde = Serdes.ByteArray();
+						valueSerde = Serdes.ByteArray();
```

Comment:
Does it have to be separate instance of Serdes.ByteArray() for key and value? Could it be shared between the two?
```
Serde serde = Serdes.ByteArray();
Serde<?> keySerde = serde;
Serde<?> valueSerde = serde;
if (bindingServiceProperties.getConsumerProperties(input). . . ) {
. . .
}
```

Reply:
Changed the logic a little bit here in the new commits.